protocol/rpcprovider/rewardserver/reward_server.go

```diff
@@ -95,6 +95,9 @@ func (rws *RewardServer) SendNewProof(ctx context.Context, proof *pairingtypes.R
 	if cuSumStored >= proof.CuSum {
 		return cuSumStored, false
 	}
+	if relayProof.Badge != proof.Badge {
```

Comment:
`relayProof.Badge != nil && proof.Badge == nil`

Reply:
its to make sure that in case we have a badge in memory its not lost with the updated proof.

Reply:
I understand, that makes sense. Fixed here: 0bec67defc5dbd2a9d1088050ec5d1e95d1d8e35